### PR TITLE
Constitutional AI recipe

### PR DESCRIPTION
**Pull request by vwxyzjn**
SFT repro: https://wandb.ai/costa-huang/huggingface/runs/4fj3uctu/overview?workspace=user-costa-huang. 
MT Bench: 6.288

DPO repro: https://wandb.ai/costa-huang/huggingface/runs/lddwve1a?workspace=user-costa-huang

Regression checking: https://wandb.ai/costa-huang/huggingface/reports/regression--Vmlldzo2Njk4NTA2

### CODE REVIEW

File: recipes/cai/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# Constitutional AI 
+
+This repo includes the recipe for training 
+
+* https://huggingface.co/HuggingFaceH4/mistral-7b-cai-alpha
```
Comment (lewtun) [inline recipes/cai/README.md:5]:
```suggestion
* https://huggingface.co/HuggingFaceH4/mistral-7b-anthropic
```

File: recipes/cai/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+* https://huggingface.co/HuggingFaceH4/mistral-7b-grok-alpha
```
Comment (lewtun) [inline recipes/cai/README.md:6]:
```suggestion
* https://huggingface.co/HuggingFaceH4/mistral-7b-grok
```

File: recipes/cai/README.md
Comment (lewtun) [inline recipes/cai/README.md:1]: `cai` is a bit cryptic IMO - can we rename the recipe folder to `constitutional-ai` please?

File: recipes/cai/README.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+## Full training examples
+
+You will require 8 GPUs (80GB of VRAM) to train the full model.
+```shell
+# Step 1 - SFT
+ACCELERATE_LOG_LEVEL=info accelerate launch --config_file recipes/accelerate_configs/deepspeed_zero3.yaml scripts/run_sft.py recipes/cai/sft/config_full.yaml
+
+# Step 2 - DPO
+ACCELERATE_LOG_LEVEL=info accelerate launch --config_file recipes/accelerate_configs/deepspeed_zero3.yaml scripts/run_dpo.py recipes/cai/dpo/config_full.yaml
```
Comment (lewtun) [inline recipes/cai/README.md:17]:
```suggestion
ACCELERATE_LOG_LEVEL=info accelerate launch --config_file recipes/accelerate_configs/deepspeed_zero3.yaml scripts/run_dpo.py recipes/constitutional-ai/dpo/config_{grok,anthropic}.yaml
```

File: recipes/cai/README.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+```
+
+
+## Advanced: generating you own dataset
+
+To generate the constitutional AI dataset, see https://github.com/huggingface/llm-swarm/tree/main/examples/hh for detailed instructions if you want build or customize the dataset.
```
Comment (lewtun) [inline recipes/cai/README.md:23]: I think `hh` is a bit cryptic as the example name in `llm_swarm` - how about we change it to `constitutional-ai` as well?

File: recipes/cai/README.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
+# Constitutional AI 
+
+This repo includes the recipe for training
```
Comment (lewtun) [inline recipes/cai/README.md:3]:
```suggestion
This repo includes the recipe for training the following models:
```

File: recipes/cai/constitution.json
Comment (lewtun) [inline recipes/cai/constitution.json:1]: Can we rename this to `constitution_anthropic.json` and also add `constitution_grok.json` with the Grok constitution tweaks?

**Review by lewtun — CHANGES_REQUESTED** [review]
Looks good, thanks for adding this recipe! I left some nits before we merge but otherwise this LGTM

File: recipes/cai/constitution.json
Comment (vwxyzjn) [reply recipes/cai/constitution.json:1]: done. also moved it to https://github.com/huggingface/llm-swarm/tree/main/examples/constitutional-ai

File: recipes/cai/README.md (hunk 4 above)
Comment (vwxyzjn) [reply recipes/cai/README.md]: done. https://github.com/huggingface/llm-swarm/tree/main/examples/constitutional-ai

**Review by lewtun — APPROVED** [review]
Thanks for iterating, let's merge!

**Pull request merged**